modules/Router.js

```diff
@@ -54,6 +59,27 @@ const Router = React.createClass({
   },
 
   componentWillMount() {
+    const { history } = this.props
+    if (isDeprecatedHistory(history)) {
+      this.setupDeprecatedHistory()
+    } else {
+      this.setupTransitionManager()
+    }
+  },
+
+  setupTransitionManager() {
+    const { history, routes, children } = this.props
+    this.transitionManager = createTransitionManager(history, createRoutes(routes || children))
+    this.transitionManager.listen((error, state) => {
+      if (error) {
+        this.handleError(error)
+      } else {
+        this.setState(state, this.props.onUpdate)
+      }
+    })
+  },
```

Comment:
Bikeshed:

I feel like the `<Router>` -> `<RouterContext>` API would be cleaner if we didn't pass the `transitionManager` into `<RouterContext>`, but instead just a `router` object.

It doesn't make a big difference right now, since we'd be passing `history` and `router` instead of `history` and `transitionManager`, but afterward we could change it so that `<Router>` only passed the `router` into `<RouterContext>`.